source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -251,7 +251,7 @@ void ShaderGraph::addDefaultGeomNode(ShaderInput* input, const GeomPropDef& geom
 void ShaderGraph::addColorTransformNode(ShaderInput* input, const ColorSpaceTransform& transform, GenContext& context)
 {
     ColorManagementSystemPtr colorManagementSystem = context.getShaderGenerator().getColorManagementSystem();
-    if (!colorManagementSystem || input->getConnection())
+    if (!input->getIsBindInput() && (!colorManagementSystem || input->getConnection()))
```

Comment:
Looking later on the input is connected as an "output socket" ShaderGraphOutputSocket if unconnected. Believe you can just check to see if the connection is of this type. Then you would not require to tag the input as a "bind input".